bootstrap.py

```diff
@@ -383,13 +414,35 @@ def install_puppet_agent():
     call_yum("install", "puppet")
     enable_service("puppet")
 
-    puppet_conf = open('/etc/puppet/puppet.conf', 'wb')
-    puppet_conf.write("""
+    puppet_major_version = get_puppet_version()
+    if puppet_major_version == 3:
+        puppet_conf_file = '/etc/puppet/puppet.conf'
+    elif puppet_major_version in [4, 5]:
+        puppet_conf_file = '/etc/puppetlabs/puppet/puppet.conf'
+    else:
+        print_error("Unsupported puppet version")
+        sys.exit(1)
+    puppet_conf = open(puppet_conf_file, 'wb')
+    if puppet_major_version == 3:
```

Comment:
I think instead of having another set of `if/elif` here, I'd put the dirs (or the whole `[main]` section) into a variable during the conf path detection, and just use the variable in one `write()`.